### PR TITLE
Fix twitter authentication provider

### PR DESCRIPTION
**Pull request by itruf**
### What does it do?

Adds access token and access token secret to twitter auth provider config and sends it to twitter auth provider.

### Why is it needed?

It's required to make auth with twitter working.

### How to test it?

Setup twitter auth provider with fixes and check it out.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

File: packages/plugins/users-permissions/admin/src/pages/Providers/utils/forms.js [hunk 1]
```diff
@@ -173,6 +183,30 @@ const forms = {
           },
         },
       ],
+      [
```
[the remark this answers is not in the document]
Comment (itruf) [reply packages/plugins/users-permissions/admin/src/pages/Providers/utils/forms.js:186]: Thank you for highlighting it. You are right, I will fix it

[the remark this answers is not in the document]
Comment (itruf) [reply packages/plugins/users-permissions/admin/src/pages/Providers/utils/forms.js]: I fixed the PR. Please, check it out

File: packages/plugins/users-permissions/admin/src/pages/Providers/utils/forms.js [hunk 2]
```diff
@@ -34,7 +34,131 @@ const secretLabel = {
...
 
+const accessToken = {
+  id: getTrad('PopUpForm.Providers.accessToken.label'),
+  defaultMessage: 'Access Token',
+};
+
+const accessTokenSecret = {
+  id: getTrad('PopUpForm.Providers.accessTokenSecret.label'),
+  defaultMessage: 'Access Token Secret',
+};
```
Comment (markkaylor) [inline packages/plugins/users-permissions/admin/src/pages/Providers/utils/forms.js:45]: I think you will need to add these translations in the en translation file.

**Review by markkaylor — COMMENTED** [review]
I'm a bit confused, why do we need the access token for twitter now? Did something change in their API? Isn't the access token something twitter responds with once the user logs in via SSO? Sorry I haven't had the time to give this a thorough review, but will try to test it later today or tomorrow.

File: packages/plugins/users-permissions/admin/src/pages/Providers/utils/forms.js (hunk 2 above)
Comment (itruf) [reply packages/plugins/users-permissions/admin/src/pages/Providers/utils/forms.js:45]: > I'm a bit confused, why do we need the access token for twitter now? Did something change in their API? Isn't the access token something twitter responds with once the user logs in via SSO? Sorry I haven't had the time to give this a thorough review, but will try to test it later today or tomorrow.

Honestly I haven't checked when it's changed and why. But twitter auth wasn't working for me, I've checked [twitter docs](https://developer.twitter.com/en/docs/authentication/oauth-1-0a) and [purest sources](https://github.com/simov/purest/blob/bc57fc4c8608c24d63f6e05fe85d409949882af8/examples/oauth-1.js#L19) and it seems that all 4 keys are required to make authentication successfully.

Comment (itruf) [reply packages/plugins/users-permissions/admin/src/pages/Providers/utils/forms.js:45]: > I think you will need to add these translations in the en translation file.

@markkaylor  Thanks, added localization

File: packages/plugins/users-permissions/server/services/providers-registry.js [hunk 3]
```diff
@@ -343,7 +345,7 @@ module.exports = () => {
       assert(typeof providerName === 'string', 'Provider name must be a string');
       assert(typeof provider === 'function', 'Provider callback must be a function');
 
-      providersCallbacks[providerName] = provider({ purest });
```
Comment (Boegie19) [inline packages/plugins/users-permissions/server/services/providers-registry.js:346]: why this change?

Comment (itruf) [reply packages/plugins/users-permissions/server/services/providers-registry.js:346]: @Boegie19 it's here by mistake, will revert it. Created another PR for this bug: https://github.com/strapi/strapi/pull/16839

Comment (itruf) [reply packages/plugins/users-permissions/server/services/providers-registry.js]: Reverted

**Comment by Boegie19** [comment]
After translations I think this one is ready to go

**Comment by itruf** [comment]
> After translations I think this one is ready to go

I've added translations to English. I think translations to other languages is not the best idea because `Access Token` is not translated usually.

**Comment by markkaylor** [comment]
I do not believe there is a bug with the twitter provider. I did the following test on `main` and everything seems to work fine. @itruf  could you try this as well:

- In the twitter developer console generate consumer keys: API KEY and API KEY SECRET
- Use these keys in strapi for the twitter provider: client id = API KEY, client secret = API KEY SECRET
- Create and configure an app. Select request email from users, set the callback uri to http://localhost:1337/api/connect/twitter/callback, set the other requested urls to anything you want with https.
- Go to localhost:1337/api/connect/twitter
- It should redirect to a twitter page asking to authorize app
- Then after authorizing it should redirect to something like: http://localhost:1337/api/auth/twitter/callback?access_token=<access_token>&oauth_token=<oauth_token>&user_id=<user_id>. There will also be a successful payload:
```
{
  jwt: '<jwt>',
  user: {
    id: 1,
    username: 'test',
    email: 'twitteremail@test.com',
    provider: 'twitter',
    confirmed: true,
    blocked: false,
    createdAt: '2023-05-31T11:57:25.158Z',
    updatedAt: '2023-05-31T11:57:25.158Z',
  },
};
```
- In Strapi you should see the user in the content-manager User model.

**Comment by itruf** [comment]
@markkaylor it produces 401 error at the last for me.

Request, curl:
```
curl --location 'https://api.casper.boo/api/auth/twitter/callback?oauth_token=gSsNUQAAAAABm6olAAABiHHvNak&oauth_verifier=FoyGcv09Z6YEAmio2Q4ouwbctkStIGOh&user_id=254181831'
```

Response:
```
{
    "data": null,
    "error": {
        "status": 400,
        "name": "ApplicationError",
        "message": "401 Unauthorized",
        "details": {}
    }
}
```

**Comment by itruf** [comment]
@markkaylor thank you for your patience and highlighting my shortcoming. You are right, it works fine without my MR.

Reason of the issue that my frontend hadn't passed `access_secret` to method `GET /api/auth/twitter/callback?access_token=token&access_secret=secret`.

PS: @Boegie19 thx for review.
